api/awsauth/api_key_repo.py

```diff
@@ -0,0 +1,65 @@
+from typing import Optional, Dict, Any
+import os
+
+import dynamo_client
+
+
+API_KEY_TABLE_NAME = os.environ["API_KEYS_TABLE"]
+API_KEY_INDEX_NAME = "apiKeys"
+
+
+class APIKeyRepo:
```

Comment:
I don't get the value of this layer of abstraction between the dynamo_client (written just for this app, entirely in our control) and auth_app.

Reply:
I've always appreciated having the layer of abstraction to separate the implementation details of using dynamo db vs more abstract requests "I would like an api key". I've also found it easier to mock out responses at a not client level.  

IMO it gives a bit more room to make the distinction between the business logic + what's actually happening behind the scenes (+ if we were to ever switch up detail of the implementation, all of the logic is contained in the repo)